src/System.Net.WebSockets.WebSocketProtocol/src/System/Net/WebSockets/ManagedWebSocketExtensions.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.WebSockets
+{
+    internal static class ManagedWebSocketExtensions
+    {
+        internal static string GetString(this UTF8Encoding encoding, Span<byte> bytes)
+        {
+            return encoding.GetString(bytes.ToArray());
+        }
+
+        internal static Task<int> ReadAsync(this Stream stream, Memory<byte> destination, CancellationToken cancellationToken)
+        {
+            if (destination.TryGetArray(out ArraySegment<byte> array))
```

Comment:
With the call sites this has in the netstandard build, will TryGetArray ever return false?  Seems like the `Memory<byte>` will have always been constructed from a byte[], in which case TryGetArray will always return true, in which case this whole implementation can be:
```C#
bool success = destination.TryGetArray(out ArraySegment<byte> array);
Debug.Assert(success);
return stream.ReadAsync(array.Array, array.Offset, array.Count, cancellationToken);
```

Reply:
The buffer passed may come from user, from the callsite in managedsocket, the buffer passed here could be the memory<byte> from WebSocketProtocol.CreateFromStream api.

Reply:
Ok